File: constrained-device-app/src/main/python/programmingtheiot/data/SystemPerformanceData.py
#####
# 
# This class is part of the Programming the Internet of Things project.
# 
# It is provided as a simple shell to guide the student and assist with
# implementation for the Programming the Internet of Things exercises,
# and designed to be modified by the student as needed.
#

from programmingtheiot.data.BaseIotData import BaseIotData
import programmingtheiot.common.ConfigConst as ConfigConst

class SystemPerformanceData(BaseIotData):
	"""
	Shell representation of class for student implementation.
	
	"""
	DEFAULT_VAL = 0.0
	DEFAULT_CPU_UTIL_DATA = 0.0
	DEFAULT_DISK_UTIL_DATA = 0.0
	DEFAULT_MEM_UTIL_DATA = 0.0
	
	def __init__(self, d = None):
		"""
		Constructor.
		
		@param d Defaults to None. The data (dict) to use for setting all parameters.
		It's provided here as a convenience - mostly for testing purposes. The utility
		in DataUtil should be used instead.
		"""
		super(SystemPerformanceData, self).__init__(name = ConfigConst.SYS_PERF_DATA, d = d)
		
		if d:
			self.cpuUtil = d['cpuUtil']
			self.diskUtil = d['diskUtil']
			self.memUtil = d['memUtil']
		else:
			self.cpuUtil = self.DEFAULT_VAL
			self.diskUtil = self.DEFAULT_VAL
			self.memUtil = self.DEFAULT_VAL
	
	"""
	Getter methods
	"""
	def getCpuUtilization(self):
		return self.cpuUtil
	
	def getDiskUtilization(self):
		return self.diskUtil
	
	def getMemoryUtilization(self):
		return self.memUtil
	
	"""
	Setter methods
	"""
	def setCpuUtilization(self, cpuUtil):
		self.cpuUtil = cpuUtil
	
	def setDiskUtilization(self, diskUtil):
		self.diskUtil = diskUtil
	
	def setMemoryUtilization(self, memUtil):
		self.memUtil = memUtil
	
	
	
	def _handleUpdateData(self, data):
		"""
		A method to update SystemPermanceData
		"""
		if data:
			self.setCpuUtilization(data.getCpuUtilization())
			self.setDiskUtilization(data.getDiskUtilization())
			self.setMemoryUtilization(data.getMemoryUtilization())

File: gateway-device-app/src/main/java/programmingtheiot/data/SystemPerformanceData.java
/**
 * This class is part of the Programming the Internet of Things project.
 * 
 * It is provided as a simple shell to guide the student and assist with
 * implementation for the Programming the Internet of Things exercises,
 * and designed to be modified by the student as needed.
 */ 

package programmingtheiot.data;

import java.io.Serializable;

import programmingtheiot.common.ConfigConst;

/**
 * Shell representation of class for student implementation.
 *
 */
public class SystemPerformanceData extends BaseIotData implements Serializable
{
	// static
	
	
	// private var's
	// CPU utilization, memory utilization, and disk utilization
	private float cpuUtil = 0.0f;
	private float memUtil = 0.0f;
	private float diskUtil = 0.0f;

	// constructors
	
	/**
	 * Default.
	 * 
	 */
	public SystemPerformanceData()
	{
		super();
		
		super.setName(ConfigConst.SYS_PERF_DATA);
	}
	
	
	// public methods
	// Getter
	public float getCpuUtilization()
	{
		return this.cpuUtil;
	}
	
	public float getDiskUtilization()
	{
		return this.diskUtil;
	}
	
	public float getMemoryUtilization()
	{
		return this.memUtil;
	}
	// Setter
	public void setCpuUtilization(float val)
	{
		this.cpuUtil = val;
	}
	
	public void setDiskUtilization(float val)
	{
		this.diskUtil = val;
	}
	
	public void setMemoryUtilization(float val)
	{
		this.memUtil = val;
	}
	
	
	// protected methods
	
	/* (non-Javadoc)
	 * @see programmingtheiot.data.BaseIotData#handleToString()
	 */
	protected String handleToString()
	{
		return null;
	}
	
	/* (non-Javadoc)
	 * @see programmingtheiot.data.BaseIotData#handleUpdateData(programmingtheiot.data.BaseIotData)
	 */
	protected void handleUpdateData(BaseIotData data)
	{
		if (data.hasError() == false) {
			this.setCpuUtilization(((SystemPerformanceData) data).getCpuUtilization());
			this.setDiskUtilization(((SystemPerformanceData) data).getDiskUtilization());
			this.setMemoryUtilization(((SystemPerformanceData) data).getMemoryUtilization());
		}
	}
	
}
